lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -54,11 +54,14 @@ def generate_token
     OAuth::AccessToken.new(oauth_consumer)
   end
 
-  def request(method, path, params = {}, data = nil, headers = {})
+  def request_uri(path)
```

Comment:
You can use `response.uri` so this isn't needed.

Reply:
The reason I introduced this method in the first place is that `#{r.uri}` would evaluate to an empty string in my tests. Let me know if you have a suggestion to fix that issue and if so I'll revert it.

Specifically, the `request` method returns something like `response = oauth_consumer.request(...)` where the oath_consumer is `OAuth::Consumer.new(...)`

So ultimately `r.uri` should be `OAuth::Consumer.new(...).request(...).uri` and what's notable to me is that when I tried it, it didn't throw any `NoMethodError` here, it simply returned empty.